shared/services/alerting/alerting.go

```diff
@@ -0,0 +1,259 @@
+package alerting
+
+import (
+	"fmt"
+	"log"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/go-openapi/strfmt"
+	apiclient "github.com/rocket-pool/smartnode/shared/services/alerting/alertmanager/client"
+	apialert "github.com/rocket-pool/smartnode/shared/services/alerting/alertmanager/client/alert"
+	"github.com/rocket-pool/smartnode/shared/services/alerting/alertmanager/models"
+	"github.com/rocket-pool/smartnode/shared/services/config"
+)
+
+const (
+	DefaultEndsAtDurationForSeverityInfo     = time.Minute * 5
+	DefaultEndsAtDurationForSeverityCritical = time.Minute * 60
+)
+
+// fetches the current alerts directly the alertmanager container/application's API.
+// If alerting/metrics are disabled, this function returns an empty array.
+func FetchAlerts(cfg *config.RocketPoolConfig) ([]*models.GettableAlert, error) {
+	// NOTE: don't log to stdout here since this method is on the "api" path and all stdout is parsed as a json "api" response.
+	if !isAlertingEnabled(cfg) {
+		// metrics are disabled, so no alerts will be fetched.
+		return make([]*models.GettableAlert, 0), nil
```

Comment:
ideally we should be able to return `nil, nil` to indicate success, no alerts

i'm guessing you segfaulted somewhere? shouldn't happen if you `len` check first, or only use `range` which is nil-safe

Reply:
Purely my ignorance of go. I actually did learn about this nil-safe iteration at some point but it was a bit after I wrote this. In other languages I always prefer empty collections instead of null/nil but I guess go has taken that classic bug into account.
Fixed.